src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -298,15 +298,14 @@ internal bool TryGetValue(int index, out int value)
             ReadOnlySpan<byte> data = _utf8Json.Span;
             ReadOnlySpan<byte> segment = data.Slice(row.Location, row.SizeOrLength);
 
-            if (Utf8Parser.TryParse(segment, out int tmp, out int consumed) &&
-                consumed == segment.Length)
+            if (!Utf8Parser.TryParse(segment, out value, out int consumed)
+                || consumed != segment.Length)
```

Comment:
I personally prefer the original code's polarity... I find the condition much easier to read without all the negation.

Reply:
Updated based on the feedback.